plugins/win-wasapi/win-wasapi.cpp

```diff
@@ -1138,17 +1138,42 @@ bool WASAPISource::ProcessCaptureData()
 			return false;
 		}
 
+		// ToDo figure out if we need to reset capture here
+		if (flags & AUDCLNT_BUFFERFLAGS_TIMESTAMP_ERROR) {
+			blog(LOG_ERROR, "[WASAPISource::ProcessCaptureData]"
+					" Timestamp error!");
+			capture->ReleaseBuffer(frames);
+			return false;
+		}
+
+		if (flags & AUDCLNT_BUFFERFLAGS_DATA_DISCONTINUITY) {
+			/* libobs should handle discontinuities fine. */
+			blog(LOG_DEBUG, "[WASAPISource::ProcessCaptureData]"
+					" Discontinuity flag is set.");
+		}
+
+		if (flags & AUDCLNT_BUFFERFLAGS_SILENT) {
+			blog(LOG_DEBUG, "[WASAPISource::ProcessCaptureData]"
+					" Silent flag is set.");
+
+			/* buffer size = frame size * number of frames
+			 * frame size = channels * sample size
+			 * sample size = 4 bytes (always float per InitFormat) */
+			uint32_t requiredBufSize = 4 * speakers * frames;
```

Comment:
`speakers` needs to be `get_audio_channels(speakers)`

Reply:
The enum is conveniently laid out such that this works anyway, but you're obviously right that this should go through that for correctness.

Reply:
No, it APPEARS conveniently laid out. Look at the end. :P